common/src/main/java/io/druid/common/utils/SerializerUtils.java

```diff
@@ -38,6 +38,34 @@
 {
   private static final Charset UTF8 = Charset.forName("UTF-8");
 
+  /**
+   * Writes the given long value into the given OutputStream, using the helperBuffer. Faster alternative to
+   * out.write(Longs.toByteArray(value)), more convenient (sometimes) than wrapping the OutputStream into {@link
+   * java.io.DataOutputStream}. The order in which the value's bytes are written to the OutputStream depends on the
+   * {@link ByteBuffer#order(ByteOrder)} of the given buffer.
+   *
+   * @param helperBuffer a heap ByteBuffer of capacity at least 8.
+   */
+  public static void writeLongToOutputStream(OutputStream out, long value, ByteBuffer helperBuffer) throws IOException
```

Comment:
Instead of switching these to be any endianness, the exact reason we got stuck in this world where we have number stored with different endianness, I would rather you keep the checks.  If you think the documentation is not enough, you could rename the methods to `writeLongBigEndian(...)`.

Either way, I just realized the signature on this method is probably incorrect.  The method depends on `.array()` actually returning something but you have no such guarantee with a `ByteBuffer` passed in.  This method should be creating its own new `helperBuffer` and using that instead of taking one as a parameter.